extensions/amp-animation/0.1/web-animations.js

```diff
@@ -234,52 +239,39 @@ export class WebAnimationRunner {
   }
 
   /**
+   * @param {!WebAnimationPlayState} playState
+   * @private
    */
-  scrollTick(pos) {
-    this.players_.forEach(player => {
-      if (player instanceof ScrollboundPlayer) {
-        player.tick(pos);
-      }
-    });
+  setPlayState_(playState) {
+    if (this.playState_ != playState) {
+      this.playState_ = playState;
+      this.playStateChangedObservable_.fire(this.playState_);
+    }
   }
 
   /**
+   * @return {!number} total duration in milliseconds.
+   * @throws {Error} If timeline is infinite.
    */
-  updateScrollDuration(newDuration) {
-    this.requests_.forEach(request => {
-      if (request.timing.ticker == Tickers.SCROLL) {
-        request.timing.duration = newDuration;
-      }
-    });
+  getTotalDuration_() {
+    let maxTotalDuration = 0;
+    for (let i = 0; i < this.requests_.length; i++) {
+      const timing = this.requests_[i].timing;
 
-    this.players_.forEach(player => {
-      if (player instanceof ScrollboundPlayer) {
-        player.onScrollDurationChanged();
-      }
-    });
-  }
+      user().assert(isFinite(timing.iterations), 'Animation has infinite ' +
+      'timeline, we can not seek to a relative position within an infinite ' +
+      'timeline. Use "time" for seekTo or remove infinite iterations');
 
-  /**
-   */
-  hasScrollboundAnimations() {
-    for (let i = 0; i < this.requests_.length; i++) {
-      if (this.requests_[i].timing.ticker == Tickers.SCROLL) {
-        return true;
+      const iteration = timing.iterations - timing.iterationStart;
```

Comment:
Heads up: this will only work with keyframe effects. The upcoming Group and Sequence effects are much harder to calculate, though, I'm sure, possible.